src/main/scala/gitbucket/core/controller/IndexController.scala

```diff
@@ -290,10 +290,16 @@ trait IndexControllerBase extends ControllerBase {
         withoutPhysicalInfo = true,
         limit = context.settings.limitVisibleRepositories
       )
-    val repositories = visibleRepositories.filter { repository =>
+
+    val repositories = {
+      context.settings.limitVisibleRepositories match {
+        case true  => getVisibleRepositories(context.loginAccount)
```

Comment:
Your fix looks to always disable `limit` option. I think it's better to modify the initial call as follows rather than make another call after checking `limitVisibleRepositories` option:
```scala
    val visibleRepositories =
      getVisibleRepositories(
        context.loginAccount,
        None,
        withoutPhysicalInfo = true,
        limit = false // fix this line
      )
```

Reply:
This url end point is required two type repository list.

```scala
gitbucket.core.search.html.repositories(
  query, 
  repositories,       // search result : always unlimited
  visibleRepositories // repo list in side bar : according to settings
  )
```

then, following implementation is also OK. however this always queries the DB twice.

```scala
get("/search") {
  val query = params.getOrElse("query", "").trim.toLowerCase

  val visibleRepositories =
    getVisibleRepositories(
    context.loginAccount,
    None,
    withoutPhysicalInfo = true,
    limit = context.settings.limitVisibleRepositories
    )

  val repositories = getVisibleRepositories(context.loginAccount).filter { repository =>
    repository.name.toLowerCase.indexOf(query) >= 0 || repository.owner.toLowerCase.indexOf(query) >= 0
  }

  gitbucket.core.search.html.repositories(query, repositories, visibleRepositories)
}
```

so, I think, previous implement is better.

Reply:
Ah, `repositories` and `visibleRepositories`. Makes sense.

Even though, `withoutPhysicalInfo` should be `false` in the second search too?

Reply:
Yes, you right.
I changed `withoutPhysicalInfo` to `true`.